pkg/authorization/registry/clusterrole/proxy.go

```diff
@@ -0,0 +1,165 @@
+package clusterrole
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metainternal "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+	restclient "k8s.io/client-go/rest"
+	rbacinternalversion "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/apis/authorization"
+	"github.com/openshift/origin/pkg/authorization/registry/util"
+)
+
+type REST struct {
+	client   restclient.Interface
+	resource schema.GroupResource
+}
+
+func NewREST(client restclient.Interface) *REST {
+	return &REST{client: client, resource: authorizationapi.Resource("clusterrole")}
+}
+
+func (s *REST) New() runtime.Object {
+	return &authorizationapi.ClusterRole{}
+}
+func (s *REST) NewList() runtime.Object {
+	return &authorizationapi.ClusterRoleList{}
+}
+
+func (s *REST) List(ctx apirequest.Context, options *metainternal.ListOptions) (runtime.Object, error) {
+	client, err := s.getImpersonatingClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	optv1 := metav1.ListOptions{}
+	if err := metainternal.Convert_internalversion_ListOptions_To_v1_ListOptions(options, &optv1, nil); err != nil {
+		return nil, err
+	}
+
+	roles, err := client.List(optv1)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &authorizationapi.ClusterRoleList{}
```

Comment:
you need to preserve the ResourceVersion of the original list (applies to all list functions)